plugins/ccgx/fu-ccgx-dmc-common.h

```diff
@@ -0,0 +1,310 @@
+	DMC_DEVICE_STATUS_PHASE2_UPDATE_FAIL_NO_VALID_KEY,
+	DMC_DEVICE_STATUS_PHASE2_UPDATE_FAIL_NO_VALID_SPI_PACKAGE,
+	DMC_DEVICE_STATUS_PHASE2_UPDATE_FAIL_RAM_INIT_FAILED,
+	DMC_DEVICE_STATUS_PHASE2_UPDATE_FAIL_FACTORY_BACKUP_FAILED,
+	DMC_DEVICE_STATUS_PHASE2_UPDATE_FAIL_NO_VALID_FACTORY_PACKAGE,
+	/* status code indicating dock FW update FAILED */
+	DMC_DEVICE_STATUS_UPDATE_FAIL = 0xFF
+} DmcDeviceStatus;
+
+/* this data type enumerates the request codes for vendor interface */
+typedef enum {
+	DMC_RQT_CODE_UPGRADE_START = 0xD0,
+	DMC_RQT_CODE_RESERV_0,
+	DMC_RQT_CODE_FWCT_WRITE,
+	DMC_RQT_CODE_IMG_WRITE,
+	DMC_RQT_CODE_RESERV_1,
+	DMC_RQT_CODE_RESERV_2,
+	DMC_RQT_CODE_DOCK_STATUS,
+	DMC_RQT_CODE_DOCK_IDENTITY,
+	/* command to reset dmc state machine of DMC */
+	DMC_RQT_CODE_RESET_STATE_MACHINE,
+	/* command to reset for online enhanced mode (no reset during update) */
+	DMC_RQT_CODE_SOFT_RESET = 0xDC,
+	/* Update Trigger command for offline mode */
+	DMC_RQT_CODE_TRIGGER = 0xDA
+} DmcRqtCode;
+
+/* this data type enumerates the opcode of triggering the download, in case of
+* 2 stage update */
+typedef enum {
+	DMC_TRIGGER_CODE_DONT_UPDATE,
+	DMC_TRIGGER_CODE_UPDATE_NOW,
+	DMC_TRIGGER_CODE_UPDATE_ON_DISCONNECT,
+	DMC_TRIGGER_CODE_UNKNOWN
+} DmcTriggerCode;
+
+/* this data type enumerates the opcode of interrupt read */
+typedef enum {
+	DMC_INT_OPCODE_FW_UPGRADE_RQT = 1,
+	DMC_INT_OPCODE_FW_UPGRADE_STATUS = 0x80,
+	DMC_INT_OPCODE_IMG_WRITE_STATUS,
+	DMC_INT_OPCODE_REENUM,
+	DMC_INT_OPCODE_FWCT_ANALYSIS_STATUS
+} DmcIntOpcode;
+
+/* this data type enumerates the fwct analysis status */
+typedef enum {
+	DMC_FWCT_ANALYSIS_STATUS_INVALID_FWCT = 0,
+	DMC_FWCT_ANALYSIS_STATUS_INVALID_DOCK_IDENTITY,
+	DMC_FWCT_ANALYSIS_STATUS_INVALID_COMPOSITE_VERSION,
+	DMC_FWCT_ANALYSIS_STATUS_AUTHENTICATION_FAILED,
+	DMC_FWCT_ANALYSIS_STATUS_INVALID_ALGORITHM
+} DmcFwctAnalysisStatus;
+
+typedef enum {
+	DMC_UPDATE_MODEL_UNKNOWN= 0,
+	/* need to trigger after updating FW */
+	DMC_UPDATE_MODEL_DOWNLOAD_TIRGGER,
+	/* need to set soft reset after updating FW */
+	DMC_UPDATE_MODEL_PENDING_RESET,
+} DmcUpdateModel;
+
+/*******************************************************************************
+ * Structures
+*******************************************************************************/
+
+/* this structure defines the fields of data returned when reading dock_identity
+ * for new firmware */
+typedef struct __attribute__((packed)) {
+	/* this field indicates both validity and structure version
+	 * 0 : invalid
+ 	 * 1 : old structure
+ 	 * 2 : new structure */
+	guint8          structure_version;
+	guint8          cdtt_version;
+	guint16         vid;
+	guint16         pid;
+	guint16         device_id;
+	gchar           vendor_string[DMC_IDENTITY_STRING_LEN];
+	gchar           product_string[DMC_IDENTITY_STRING_LEN];
+	guint8          custom_meta_data_flag;
+	/* model feild indicates the type of the firmware upgrade status
```

Comment:
field

Reply:
Fixed it on commit today (ccgx: fix code alignment, modify code for review).